rs/pocket_ic_tests/src/lib.rs

```diff
@@ -0,0 +1,347 @@
+pub use ic_state_machine_tests::{
+    CanisterId, Cycles, ErrorCode, PrincipalId, StateMachineConfig, StateMachineNode,
+    StateMachineStateDir, SubmitIngressError, Time, UserError, WasmResult,
+};
+
+use ic_registry_routing_table::RoutingTable;
+use ic_registry_subnet_type::SubnetType;
+use ic_state_machine_tests::{
+    CanisterInstallMode, CanisterSettingsArgs, CanisterStatusResultV2, MessageId, SubnetId,
+};
+use std::time::{Duration, SystemTime};
+
+#[derive(Default)]
+pub struct StateMachine {
+    sm: ic_state_machine_tests::StateMachine,
+}
+
+#[derive(Default)]
+pub struct StateMachineBuilder {
+    sm_builder: ic_state_machine_tests::StateMachineBuilder,
+}
+
+impl StateMachineBuilder {
+    // Tricky!!!
+    // subnet_config.scheduler_config.max_instructions_per_slice = MAX_INSTRUCTIONS_PER_SLICE
+    // rate_limiting_of_instructions: FlagStatus::Disabled
```

Comment:
work-around: a few ticks in between install_code messages

Reply:
those ticks could be performed by a library function upon receiving an error due to being rate-limited

Reply:
or maybe one can just use `.with_nonmainnet_features`